docs/learn.md

```diff
@@ -126,10 +126,281 @@ OCaml. Some examples are the upgrade of the compiler version used by Melange to
+manual](https://opam.ocaml.org/doc/Manual.html) and [FAQ
+page](https://opam.ocaml.org/doc/FAQ.html).
+
+### opam for Melange developers
+
+Before diving into specifics about using opam, there are the two relevant
+differences between opam and npm that are worth mentioning.
+
+**1. One version of each package**
+
+At any given time, any opam switch can only *at most* a single version of a
+package. This is known as a flat dependency graph, and some package managers in
+other languages (like [Bower](https://bower.io/)) follow a similar approach.
+
+A flat dependency graph means that, for example, it is impossible to have two
+versions of [`reason-react`](https://github.com/reasonml/reason-react/)
+installed in the same project. This avoids some headaches when one inadvertently
+installs two versions of a dependency. Also, and specifically for Melange, it
+helps keep the resulting JavaScript bundle lean and reduce page load for
+browser-based applications.
+
+On the other hand, upgrading your project dependencies to more recent versions
+might become tricky. Due to the restriction where only one version of a package
+can be installed, there is a higher chance for conflicts between the constraints
+of the transitive dependencies. If opam cannot find a solution, these conflicts
+need to be solved manually. This generally involves updating the conflicting
+dependency to make it compatible with a newer version of Melange or a transient
+dependency.
+
+**2. A source-based package manager for a compiled language**
+
+opam distributes just the source code of the packages and leaves the compilation
+step to a build phase that runs when consuming them, after they have been
+fetched. As the package manager for a compiled language like OCaml, opam has
+first-class support for this build step. Every package must tell opam how it
+should be built, and the way to do this is by using the [`build`
+field](https://opam.ocaml.org/doc/Manual.html#opamfield-build) in the package
+`.opam` file.
+
+This is different from how npm handles packages. Because npm has been designed
+for JavaScript (an interpreted language) having a build step makes no sense.
+Whenever any project or community tries to use npm to distribute software that
+includes code written in compiled languages, the burden to distribute pre-built
+binaries is imposed on library authors, like [the node-sass
+example](https://github.com/sass/node-sass/issues/1589) shows.
+
+As Melange relies heavily on OCaml packages for the compilation step (either
+PPXs, linters, instrumentation, or any other build-time package), using opam
+provides access to these tools without library authors having to care about
+creating and distributing pre-built versions of their packages.
+
+---
+
+Let's go now through the most common actions with opam when working on Melange
+projects. The following guide is based on the amazing [opam for npm/yarn
+users](http://ocamlverse.net/content/opam_npm.html) guide by Louis
```

Comment:
I left it out purposely out of respect for privacy, because I noticed it was not mentioned on [his GH profile](https://github.com/Khady) nor the [original OCamlverse post](http://ocamlverse.net/content/opam_npm.html).